### PR TITLE
Add incubation period and improve color coding

### PR DESCRIPTION
**Pull request by marcopierobon**
Added the following features:

allow the selection of the percentage of people using the app
map that percentage with a smaller inner dot
allow an incubation time before the someone who's infected can spread the disease further

### CODE REVIEW

File: src/Ball.js [hunk 1]
```diff
@@ -48,22 +50,46 @@ export class Ball {
         RUN.results[STATES.infected]--
         RUN.results[STATES.recovered]++
         this.hasMovement = true
+        if (this.vx > 0) {
```
Comment (brushtyler) [inline src/Ball.js]: please, add a comment explaining the following new block of code.

File: src/Ball.js [hunk 2]
```diff
@@ -48,22 +50,46 @@ export class Ball {
...
+          this.vx = Math.min(SPEED, this.vx * 4)
+        } else if (this.vx === 0) {
+          this.vx = this.sketch.random(-1, 1) * SPEED
+        } else {
+          this.vx = Math.max(-SPEED, this.vx * 4)
+        }
+        if (this.vy > 0) {
+          this.vy = Math.min(SPEED, this.vy * 4)
```
Comment (brushtyler) [inline src/Ball.js]: Has the `4` factor any particular reason? Why `4` and not `5` or `3`?

File: src/Ball.js [hunk 3]
```diff
@@ -48,22 +50,46 @@ export class Ball {
...
+        } else if (this.vx === 0) {
```
Comment (brushtyler) [inline src/Ball.js]: typo, it should be `this.vy === 0`

File: src/Ball.js [hunk 4]
```diff
@@ -74,15 +100,27 @@ export class Ball {
...
-          this.state = otherBall.state = STATES.infected
-          RUN.results[STATES.infected]++
-          RUN.results[STATES.well]--
+        if (this.state === STATES.infected || otherBallState === STATES.infected) {
+          if (this.state === STATES.well) {
+            this.state = STATES.incubating
+            RUN.results[STATES.incubating]++
+            RUN.results[STATES.well]--
+            if (this.hasAppInstalled && otherBall.hasAppInstalled) {
+              this.hasMovement = false
```
Comment (brushtyler) [inline src/Ball.js:120]: please, add a comment to explain why the person stops moving.

File: src/index.html [hunk 5]
```diff
@@ -215,7 +217,7 @@ <h1>COVID-19 Spread Simulator 🦠</h1>
   </main>
   
   <footer>
-    Based on <a target="_blank" rel="noopener nofollow" href='https://www.washingtonpost.com/graphics/2020/world/corona-simulator/'>Washington Post article</a> | <a href='https://github.com/noiapp/covid-19-spread-simulator' target="_blank" rel="noopener nofollow">Source Code</a> | By <a href='https://midu.dev'>@midudev</a>
+    Based on <a target="_blank" rel="noopener nofollow" href='https://www.washingtonpost.com/graphics/2020/world/corona-simulator/'>Washington Post article</a> | <a href='https://github.com/midudev/covid-19-spread-simulator' target="_blank" rel="noopener nofollow">Source Code</a> | By <a href='https://midu.dev'>@midudev</a> | Extended by <a href='https://www.linkedin.com/in/marco-pierobon-a82ab646/'>Marco Pierobon</a>
```
Comment (brushtyler) [inline src/index.html]: Consider to move attribution to a new `CONTRIBUTORS.md` file, where you can store your info alongside a summary of the changes you did. IMHO the right way would be a link to the CONTRIBUTOR file.

**Review by brushtyler — CHANGES_REQUESTED** [review]

File: src/Ball.js (hunk 1 above)
Comment (marcopierobon) [reply src/Ball.js]: Resolved with a private method with a meaningful name

File: src/Ball.js (hunk 3 above)
Comment (marcopierobon) [reply src/Ball.js]: Well spotted

File: src/Ball.js (hunk 4 above)
Comment (marcopierobon) [reply src/Ball.js:120]: Resolved with a private method with a meaningful name and a comment

File: src/index.html (hunk 5 above)
Comment (marcopierobon) [reply src/index.html]: Removed - we'll discuss attribution later, not a priority

File: src/index.html [hunk 6]
```diff
@@ -217,7 +217,7 @@ <h1>COVID-19 Spread Simulator 🦠</h1>
   </main>
   
   <footer>
-    Based on <a target="_blank" rel="noopener nofollow" href='https://www.washingtonpost.com/graphics/2020/world/corona-simulator/'>Washington Post article</a> | <a href='https://github.com/midudev/covid-19-spread-simulator' target="_blank" rel="noopener nofollow">Source Code</a> | By <a href='https://midu.dev'>@midudev</a> | Extended by <a href='https://www.linkedin.com/in/marco-pierobon-a82ab646/'>Marco Pierobon</a>
+    Based on <a target="_blank" rel="noopener nofollow" href='https://www.washingtonpost.com/graphics/2020/world/corona-simulator/'>Washington Post article</a> | <a href='https://github.com/noiapp/covid-19-spread-simulator' target="_blank" rel="noopener nofollow">Source Code</a> | By <a href='https://midu.dev'>@midudev</a>	    Based on <a target="_blank" rel="noopener nofollow" href='https://www.washingtonpost.com/graphics/2020/world/corona-simulator/'>Washington Post article</a> | <a href='https://github.com/midudev/covid-19-spread-simulator' target="_blank" rel="noopener nofollow">Source Code</a> | By <a href='https://midu.dev'>@midudev</a>
```
Comment (brushtyler) [inline src/index.html:220]: Duplicated "Based on ... | By ..." on this line.

File: src/options.js [hunk 7]
```diff
@@ -61,6 +61,7 @@ export const SPEED = 1
 export const TOTAL_TICKS = 1600
 export const TICKS_TO_RECOVER = 500
 export const TICKS_TO_INCUBATE = 100
+export const DECELERATION_WHEN_SICK = 100
```
Comment (brushtyler) [inline src/options.js:64]: Change the name of the option `DECELERATION_WHEN_SICK` as it's not used when sick, only when recovered.

File: src/Ball.js [hunk 8]
```diff
@@ -109,15 +114,18 @@ export class Ball {
             this.state = STATES.incubating
             RUN.results[STATES.incubating]++
             RUN.results[STATES.well]--
-            if (this.hasAppInstalled && otherBall.hasAppInstalled) {
+
+            if (this.hasTheAppInformedOfAContactWithAnInfectedPerson(otherBall)) {
```
Comment (brushtyler) [inline src/Ball.js:118]: The previous code was easier to be read (and understood) than this very long function name.
I would suggest a shorter name for it, e.g. `awareToBeInfectedBy`

File: src/Ball.js [hunk 9]
```diff
@@ -80,6 +68,23 @@ export class Ball {
     }
   }
 
+  accelerateBackToHealthyLevel () {
+    if (this.vx > 0) {
+      this.vx = Math.min(SPEED, this.vx * DECELERATION_WHEN_SICK)
```
Comment (brushtyler) [inline src/Ball.js:73]: This code has been changed from `this.vx * 4` to `this.vx * 100`.
I'm not sure this is the expected behavior. I would expect the previous factor, while 100 seems to me a scalar value.

**Review by brushtyler — CHANGES_REQUESTED** [review]

File: src/Ball.js (hunk 8 above)
Comment (marcopierobon) [reply src/Ball.js:118]: Yes, although I advise to always use the prefix is/has when a boolean is expected. isAwareToBeInfected will do

File: src/Ball.js (hunk 9 above)
Comment (marcopierobon) [reply src/Ball.js:73]: Fair point, at the moment SPEED would always be taken. Will roll it back

File: src/options.js (hunk 7 above)
Comment (marcopierobon) [reply src/options.js:64]: Yep

File: src/index.html (hunk 6 above)
Comment (marcopierobon) [reply src/index.html:220]: Yes, I had seen this and fixed it locally

File: src/Ball.js (hunk 8 above)
Comment (brushtyler) [reply src/Ball.js:118]: Change moved forward to next PR.

File: src/options.js (hunk 7 above)
Comment (brushtyler) [reply src/options.js:64]: Change moved forward to next PR.

File: src/index.html (hunk 6 above)
Comment (brushtyler) [reply src/index.html:220]: Change moved forward to next PR.

File: src/Ball.js (hunk 9 above)
Comment (brushtyler) [reply src/Ball.js:73]: Change moved forward to next PR.

**Review by brushtyler — APPROVED** [review]

**Comment by brushtyler** [comment]
Requested changed moved forward to next PR.

**Pull request merged**